minio/minio
# Generalize the event store using go generics

## Description

This PR makes the event store generic so that it can be used by other targets like logger webhook targets, audit webhook targets and audit kafka targets etc.

## Motivation and Context

We will be implementing `queue_dir` support for audit and logger targets soon. For this, we need to generalize the current event store.

## How to test this PR?

Just some sanity tests to make sure the `queue_dir` functionality in event notifications work fine.

- Configure any notification target with `queue_dir` enabled
- Test event store functionality. ie, take down the target, send events and check if the events are properly queued in the provided `queue_dir` and then bring the target up and check if the events are getting replayed properly.

## Types of changes

## Checklist:
